model_compression_toolkit/data_generation/pytorch/image_operations.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2024 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import numpy as np
+import torch
+import torch.nn.functional as F
+from model_compression_toolkit.core.pytorch.pytorch_device_config import get_working_device
+from torchvision.transforms import Normalize
+
+def create_valid_grid(means, stds) -> torch.Tensor:
+    """
+    Create a valid grid for image normalization.
+
+    Returns:
+        torch.Tensor: The valid grid for image normalization.
+    """
+    # Image valid grid
+    pixel_grid = torch.from_numpy(np.array(list(range(256))).repeat(3).reshape(-1, 3)).float()
+    valid_grid = Normalize(mean=means, std=stds)(pixel_grid.transpose(1, 0)[None, :, :, None]).squeeze().to(
+        get_working_device())
+    return valid_grid
+
+
+class Smoothing(torch.nn.Module):
+    """
+    A PyTorch module for applying Gaussian smoothing to an image.
+    """
+
+    def __init__(self, size: int = 3, sigma: float = 1.25, kernel: torch.Tensor = None):
+        """
+        Initialize the Smoothing module.
+
+        Args:
+            size (int): The size of the Gaussian kernel.
+            sigma (float): The standard deviation of the Gaussian kernel.
+            kernel (torch.Tensor, optional): Precomputed Gaussian kernel. If None, it will be created.
+        """
+        super().__init__()
+        if kernel is None:
```

Comment:
if kernel is provided you don't need size and sigma?

Reply:
yes